modules/globebrowsing/other/pixelbuffercontainer.inl

```diff
@@ -0,0 +1,131 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2017                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+namespace openspace {
+namespace globebrowsing {
+
+template <class KeyType>
+PixelBufferContainer<KeyType>::PixelBufferContainer(size_t numBytesPerBuffer,
+    PixelBuffer::Usage usage, size_t numPixelBuffers)
+    : _usage(usage)
+{
+    for (size_t i = 0; i < numPixelBuffers; ++i) {
+        _pixelBuffers.push_back(std::make_unique<PixelBuffer>(numBytesPerBuffer, _usage));
+    }
+}
+
+template <class KeyType>
+void* PixelBufferContainer<KeyType>::mapBuffer(KeyType key, GLenum access) {
+    typename std::map<KeyType, int>::iterator iter = _indexMap.find(key);
```

Comment:
Ah...  I think it might be because by the time the code is being parsed, `KeyType` is still unknown, so you need to tell the parser that `std::map<KeyType, int>::iterator` is a type, rather than, for example a `static` constant.

cf:
```
typename <int i>
struct A {};

typename <>
struct A<0> {
  using foo = float;
};

template <>
struct A<1> {
   const int foo = 2;
}
```